service/mantle/ups/UpsServices.xml

```diff
@@ -47,4 +47,47 @@
         </actions>
     </service>
 
+    <service verb="get" noun="PackageInfo">
+        <implements service="mantle.shipment.CarrierServices.get#AutoPackageInfo"/>
+        <actions>
+            <!--Get shipmentIds from ShipmentAndItemSource using orderId and orderItemSeqId of itemInfoList-->
+            <set field="shipmentIdsList" from="new HashSet()"/>
+            <iterate list="itemInfoList" entry="itemInfoMap">
+                <entity-find-one entity-name="mantle.shipment.ShipmentAndItemSource" value-field="shipmentSourceList">
+                    <field-map field-name="orderId" from="itemInfoMap.orderId"/>
+                    <field-map field-name="orderItemSeqId" from="itemInfoMap.orderItemSeqId"/>
+                </entity-find-one>
+                <script>shipmentIdsList.add(shipmentSourceList.shipmentId)</script>
+            </iterate>
+
+            <set field="packageInfoList" from="[]"/>
+
+            <iterate list="shipmentIdsList" entry="shipment">
+                <entity-find entity-name="mantle.shipment.ShipmentPackage" list="packageList">
+                    <econdition field-name="shipmentId" from="shipment"/>
+                    <order-by field-name="shipmentPackageSeqId"/>
+                </entity-find>
+
+                <entity-find entity-name="mantle.shipment.ShipmentPackageContent" list="shipmentPackageContentList">
+                    <econdition field-name="shipmentId" from="shipment"/>
+                    <order-by field-name="productId"/>
+                </entity-find>
+
+                <iterate list="packageList" entry="pkg">
+                    <filter-map-list list="shipmentPackageContentList" to-list="curContentList">
```

Comment:
Please add some comments for the code regarding how we are preparing the list. Documentation of code is very important.